### ISSUE TITLE
Changing runtime in a lambda function makes it lose triggers

### ISSUE THREAD

**Issue by brikis98** [hashicorp/terraform#12181]
### Terraform Version

Terraform v0.8.6

### Affected Resource(s)

- aws_lambda_function
- aws_sns_topic_subscription
- aws_lambda_permission

### Terraform Configuration Files

```hcl
resource "aws_lambda_function" "foo" {
  function_name = "foo"
  filename = "foo.zip"
  source_code_hash = "12345"
  handler = "index.handler"
  role = "${aws_iam_role.drain_ecs_instances_lambda_job.arn}"

  runtime = "nodejs4.3"
}

resource "aws_sns_topic" "foo_notifications" {
  name = "foo-notifications"
}

resource "aws_sns_topic_subscription" "foo_subscription" {
  topic_arn = "${aws_sns_topic.foo_notifications.arn}"
  protocol  = "lambda"
  endpoint  = "${aws_lambda_function.foo.arn}"
}

resource "aws_lambda_permission" "allow_sns" {
  statement_id = "allow-sns"
  action = "lambda:InvokeFunction"
  function_name = "${aws_lambda_function.foo.function_name}"
  principal = "sns.amazonaws.com"
  source_arn = "${aws_sns_topic.foo_notifications.arn}"
}
```

### Steps to Reproduce

1. `terraform apply`
1. Check that the `foo` function gets created properly and has an SNS trigger registered.
1. Change `aws_lambda_function` to a different zip file which contains Python code and change the runtime to `python2.7`.
1. `terraform apply`
1. Check that the `foo` function gets recreated and still has the SNS trigger registered.

### Expected Behavior

The `foo` function has the trigger registered before and after the `runtime` change.

### Actual Behavior

It has the trigger before, but not after.

**Comment by stack72** [comment]
Hi @brikis98 

I am currently looking at a fix for this - IMO, I don't believe it is necessary to ForceNew resource on change on runtime, the API allows a change of runtime

Agree?

Paul

**Comment by brikis98** [comment]
@stack72 That seems like a reasonable approach.

**Comment by brikis98** [comment]
Thx!

**Comment by stack72** [comment]
ok, the PR is in and passes :) I hope this will be available for you in Terraform 0.8.8

P.

**Comment by brikis98** [comment]
Excellent, thank you!
